auth/auth/exceptions.py

```diff
@@ -0,0 +1,52 @@
+from aiohttp import web
+
+
+class AuthUserError(Exception):
+    def __init__(self, message, severity):
+        super().__init__(message)
+        self.message = message
+        self.ui_error_type = severity
+
+    def http_response(self):
+        return web.HTTPBadRequest(reason=self.message)
+
+
+class EmptyLoginID(AuthUserError):
+    def __init__(self, username):
+        super().__init__(f"Login id for user '{username}' must be a non-empty string.", 'error')
+
+
+class DuplicateLoginID(AuthUserError):
+    def __init__(self, username, login_id):
+        super().__init__(f"Login id '{login_id}' already exists for user '{username}'.", 'error')
+
+
+class DuplicateUsername(AuthUserError):
+    def __init__(self, username, login_id):
+        super().__init__(f"Username '{username}' already exists with login id '{login_id}'.", 'error')
+
+
+class InvalidUsername(AuthUserError):
+    def __init__(self, username):
+        super().__init__(f"Invalid username '{username}'. Must be a non-empty alphanumeric string.", 'error')
+
+
+class InvalidType(AuthUserError):
+    def __init__(self, field_name, input, expected_type):
+        super().__init__(f"Expected '{field_name}' is type {expected_type}. Found type {type(input)}", 'error')
+
+
+class MultipleUserTypes(AuthUserError):
+    def __init__(self, username):
+        super().__init__(f"User '{username}' cannot be both a developer and a service account.", 'error')
+
+
+class MultipleExistingUsers(AuthUserError):
+    def __init__(self, username, login_id):
+        super().__init__(
+            f"User with user name '{username}' and login id '{login_id}' appear multiple times in the database.",
```

Comment:
text seems a bit odd. maybe "Multiple users with user name ... appear in the database.